src/apps/stable/routes/search.tsx

```diff
@@ -7,18 +7,42 @@ import SearchResults from '../../../components/search/SearchResults';
 import SearchSuggestions from '../../../components/search/SearchSuggestions';
 import LiveTVSearchResults from '../../../components/search/LiveTVSearchResults';
 import globalize from '../../../scripts/globalize';
+import { history } from '../../../components/router/appRouter';
+
+function usePrevious(value: string) {
+    const ref = useRef<string>('');
+    useEffect(() => {
+        ref.current = value;
+    });
+    return ref.current;
+}
 
 const Search: FunctionComponent = () => {
-    const [ query, setQuery ] = useState<string>();
     const [ searchParams ] = useSearchParams();
+    const [ query, setQuery ] = useState<string>(searchParams.get('query') || '');
+    const prevQuery = usePrevious(query);
```

Comment:
The ability to know what `query` was previously set to allows us to detect which direction the change came from (a change to the URL or a change to the input field).